repos/system_upgrade/el8toel9/actors/nisscanner/libraries/nisscan.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+from leapp.libraries.stdlib import api
+from leapp.models import NISConfig
+
+PACKAGES_NAMES = ('ypserv', 'ypbind')
+YPBIND_CONF_FILE = '/etc/yp.conf'
+YPSERV_DIR_PATH = '/var/yp'
+YPSERV_DEFAULT_FILES = ('binding', 'Makefile', 'nicknames')
+
+
+class NISScanLibrary(object):
+    """
+    Helper library for NISScan actor.
+    """
+
+    def client_has_non_default_configuration(self):
+        """
+        Check for any significant ypbind configuration lines in .conf file.
+        """
+        if not os.path.isfile(YPBIND_CONF_FILE):
+            return False
+
+        # Filter whitespaces and empty lines
+        with open(YPBIND_CONF_FILE) as f:
+            lines = [line.strip() for line in f.readlines() if line.strip()]
+
+        for line in lines:
+            # Cheks for any valid configuration entry
+            if not line.startswith('#'):
+                return True
```

Comment:
So a non default configuration contains comments, and there's nothing else that can be changed?
Wouldn't it better to ask rpm -V for the file to check if it is default?

Reply:
In this actor it's ok to do that, the scanner may do that.

Reply:
I don't really like the idea of executing rpm command and I cannot find docs for rpm bindings. I think that my solution is slightly better. Someone can add empty line to config which is meaningless and that would result with outcome:

 "Hey someone change it so it's configured"

if we use rpm to check that file.